core/src/main/java/de/bwaldvogel/mongo/oplog/CollectionBackedOplog.java

```diff
@@ -0,0 +1,45 @@
+package de.bwaldvogel.mongo.oplog;
+
+import java.time.Clock;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import de.bwaldvogel.mongo.MongoCollection;
+import de.bwaldvogel.mongo.bson.BsonTimestamp;
+import de.bwaldvogel.mongo.bson.Document;
+import io.netty.channel.Channel;
+
+public class CollectionBackedOplog extends AbstractOplog {
+
+    private final MongoCollection<Document> collection;
+    public CollectionBackedOplog(Clock clock, MongoCollection<Document> collection) {
+        super(clock);
+        this.collection = collection;
+    }
+
+    @Override
+    public void handleInsert(Channel channel, String databaseName, Document query) {
+        LocalDateTime now = LocalDateTime.now(clock);
+        List<Document> documents = (List<Document>) query.get("documents");
+        List<Document> oplogDocuments = documents.stream().map(d ->
+            new OplogDocument()
+                .withTimestamp(new BsonTimestamp(now.toEpochSecond(ZoneOffset.UTC)))
```

Comment:
Use [Clock.millis](https://docs.oracle.com/javase/8/docs/api/java/time/Clock.html#millis--)

Reply:
I used 
`Instant instant = clock.instant();`
`instant.toEpochMilli()`

instead as the clock docs reads

>  Most applications should avoid this method and use {@link Instant} to represent
     an instant on the time-line rather than a raw millisecond value.

Reply:
Up to you. Default implementation uses `instant()` anyway:

```java
public long millis() {
        return instant().toEpochMilli();
    }
```